src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -1225,60 +1201,7 @@ public virtual int LastIndexOf(string source, string value, int startIndex, int
         }
 
         public virtual int LastIndexOf(string source, char value, int startIndex, int count, CompareOptions options)
-        {
-            if (source == null)
-            {
-                throw new ArgumentNullException(nameof(source));
-            }
-            // Validate CompareOptions
-            // Ordinal can't be selected with other flags
-            if ((options & ValidIndexMaskOffFlags) != 0 &&
-                (options != CompareOptions.Ordinal) &&
-                (options != CompareOptions.OrdinalIgnoreCase))
-            {
-                throw new ArgumentException(SR.Argument_InvalidFlag, nameof(options));
-            }
-
-            // Special case for 0 length input strings
-            if (source.Length == 0 && (startIndex == -1 || startIndex == 0))
-            {
-                return -1;
-            }
-
-            // Make sure we're not out of range
-            if (startIndex < 0 || startIndex > source.Length)
-            {
-                throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_Index);
-            }
-
-            // Make sure that we allow startIndex == source.Length
-            if (startIndex == source.Length)
-            {
-                startIndex--;
-                if (count > 0)
-                {
-                    count--;
-                }
-            }
-
-            // 2nd have of this also catches when startIndex == MAXINT, so MAXINT - 0 + 1 == -1, which is < 0.
-            if (count < 0 || startIndex - count + 1 < 0)
-            {
-                throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_Count);
-            }
-
-            if (options == CompareOptions.OrdinalIgnoreCase)
-            {
-                return source.LastIndexOf(value.ToString(), startIndex, count, StringComparison.OrdinalIgnoreCase);
-            }
-
-            if (GlobalizationMode.Invariant)
-            {
-                return InvariantLastIndexOf(source, char.ToString(value), startIndex, count, (options & (CompareOptions.IgnoreCase | CompareOptions.OrdinalIgnoreCase)) != 0);
-            }
-
-            return LastIndexOfCore(source, value.ToString(), startIndex, count, options);
-        }
+            => LastIndexOf(source, value.ToString(), startIndex, count, options);
```

Comment:
>value.ToString() [](start = 35, length = 16)

we don't have the API overload that take a span instead of string? I am asking because we can avoid value.ToString() by using span

Reply:
Fixed in most recent iteration.